arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -13,3 +13,18 @@ config ARM_MPU
 	default n
 	help
 	  MCU has ARM MPU
+
+config NXP_MPU
+	bool "NXP MPU Support"
+	depends on CPU_HAS_MPU
+	depends on SOC_FAMILY_KINETIS
+	default n
+	help
+	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
```

Comment:
Hrm.. now that I look, there isn't in a use-case in Zephyr for enabling CONFIG_FLASH that isn't write/erase centric.   Prior to adding a CONFIG_* option to adjust MPU settings, I was thinking of a dynamic MPU hook which could be triggered via flash_write_protection_set() calls.  But, I'm not sure if we can do that in a generic way across MPU implementations.  But the side effect of that method would be no additional CONFIG options.